quantus/metrics/robustness/relative_output_stability.py

```diff
@@ -0,0 +1,314 @@
+        x_batch: np.ndarray,
+        y_batch: np.ndarray,
+        model_predict_kwargs: Optional[Dict[str, ...]] = None,
+        explain_func: Optional[Callable] = None,
+        explain_func_kwargs: Optional[Dict[str, ...]] = None,
+        a_batch: Optional[np.ndarray] = None,
+        device: Optional[str] = None,
+        softmax: bool = False,
+        channel_first: bool = True,
+        batch_size: int = 64,
+        **kwargs,
+    ) -> List[float] | float:
+        """
+        Parameters
+        ----------
+        model: tf.keras.Model, torch.nn.Module
+            A torch or tensorflow model that is subject to explanation.
+        x_batch: np.ndarray
+            4D tensor representing batch of input images
+        y_batch: np.ndarray
+            1D tensor, representing predicted labels for the x_batch.
+        model_predict_kwargs: dict, optional
+            Keyword arguments to be passed to the model's predict method.
+        explain_func: callable, optional
+            Function used to generate explanations.
+        explain_func_kwargs: dict, optional
+            Keyword arguments to be passed to explain_func on call.
+        a_batch: np.ndarray, optional
+            4D tensor with pre-computed explanations for the x_batch.
+        device: str, optional
+            Device on which torch should perform computations.
+        softmax: boolean, optional
+            Indicates whether to use softmax probabilities or logits in model prediction.
+            This is used for this __call__ only and won't be saved as attribute. If None, self.softmax is used.
+        channel_first: boolean, optional
+            Indicates of the image dimensions are channel first, or channel last.
+            Inferred from the input shape if None.
+        kwargs:
+            not used, deprecated
+        Returns
+        -------
+        relative output stability: float, np.ndarray
+            float in case `return_aggregate=True`, otherwise np.ndarray of floats
+
+
+        For each image `x`:
+         - generate `num_perturbations` perturbed `xs` in the neighborhood of `x`
+         - find `xs` which results in the same label
+         - Compute explanations `e_x` and `e_xs`
+         - Compute relative input output objective, find max value with respect to `xs`
+         - In practise we just use `max` over a finite `xs_batch`
+        """
+        result = super().__call__(
+            model=model,
+            x_batch=x_batch,
+            y_batch=y_batch,
+            explain_func=explain_func,
+            explain_func_kwargs=explain_func_kwargs,
+            a_batch=a_batch,
+            device=device,
+            softmax=softmax,
+            channel_first=channel_first,
+            model_predict_kwargs=model_predict_kwargs,
+            s_batch=None,
+            batch_size=batch_size,
+        )
+        return list(result) if isinstance(result, Iterable) else float(result)  # noqa
```

Comment:
for readability, make it several lines (matter or taste but still)

Reply:
Done for all 3 metrics.